pystac/collection.py

```diff
@@ -760,6 +766,50 @@ def delete_asset(self, key: str) -> None:
 
         del self.assets[key]
 
+    def make_asset_hrefs_relative(self) -> Collection:
+        """Modify each asset's HREF to be relative to this collection's self HREF.
+
+        Returns:
+            Collection: self
+        """
+        self_href = None
+        for asset in self.assets.values():
+            href = asset.href
+            if is_absolute_href(href):
+                if self_href is None:
+                    self_href = self.get_self_href()
+                    if self_href is None:
+                        raise STACError(
+                            "Cannot make asset HREFs relative "
+                            "if no self_href is set."
+                        )
+                asset.href = make_relative_href(asset.href, self_href)
+        return self
+
+    def make_asset_hrefs_absolute(self) -> Collection:
+        """Modify each asset's HREF to be absolute.
+
+        Any asset HREFs that are relative will be modified to absolute based on this
+        collection's self HREF.
+
+        Returns:
+            Collection: self
+        """
+        self_href = None
+        for asset in self.assets.values():
+            href = asset.href
+            if not is_absolute_href(href):
+                if self_href is None:
+                    self_href = self.get_self_href()
+                    if self_href is None:
+                        raise STACError(
+                            "Cannot make relative asset HREFs absolute "
+                            "if no self_href is set."
+                        )
+                asset.href = make_absolute_href(asset.href, self_href)
```

Comment:
There could be something I'm missing here, but I think this can be de-nested like this:
```suggestion
        self_href = self.get_self_href()
        if self_href is None:
            raise STACError(
                "Cannot make relative asset HREFs absolute "
                "if no self_href is set."
            )
        for asset in self.assets.values():
            href = asset.href
            if not is_absolute_href(href):
                asset.href = make_absolute_href(asset.href, self_href)
```

Reply:
oh yeah I should have mentioned that I just copied this over from the Item class. I like yours though. I'll change it in both places.

Reply:
I kept the error raising within the assets loop since we probably don't need to raise an error if there are no assets that need to be changed.